data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkOperationWrapper.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.opensearch;
+
+import org.opensearch.client.opensearch.core.bulk.BulkOperation;
+import org.opensearch.dataprepper.model.event.EventHandle;
+
+public class BulkOperationWrapper {
+    private final EventHandle eventHandle;
+    private final BulkOperation bulkOperation;
+
+    public BulkOperationWrapper(final BulkOperation bulkOperation) {
+        this.bulkOperation = bulkOperation;
+        this.eventHandle = null;
+    }
+
+    public BulkOperationWrapper(final BulkOperation bulkOperation, final EventHandle eventHandle) {
+        this.bulkOperation = bulkOperation;
```

Comment:
We should enforce BulkOperation cannot be null. Otherwise, we may encounter a bunch of NPE's where we leverage this object.

Sample Reference:
```
 anyDocument = bulkOperation.getBulkOperation().index().document();
```